chalk-integration/src/lowering.rs

```diff
@@ -236,6 +246,12 @@ impl LowerProgram for Program {
                     trait_ids.insert(type_kind.name, id);
                     trait_kinds.insert(id, type_kind);
                 }
+                Item::OpaqueTyDefn(defn) => {
```

Comment:
We seem a bit inconsistent in whether we are calling this "opaque ty" or "impl trait" here -- is there a reason to use different terms in different places? I'm somewhat indifferent about which we use. I have a vague preference for `impl Trait` (more tightly tied to Rust syntax) but then rustc is using (I think) opaque type.

Reply:
Yeah, we chose `opaque type` syntax in the AST, so I'm using that around the ast types. Would you prefer it to be called ImplTrait in the code everywhere, while still using `opaque type` syntax?

Reply:
After discussing this, we decided to go with "opaque type" in most places, and also rename ImplTraitDatum to OpaqueTyDatum, for example